nucypher/network/nodes.py

```diff
@@ -1310,75 +1291,64 @@ def timestamp_bytes(self):
+            last_seen = self.last_seen.iso8601()
         except AttributeError:
-            last_seen = str(node.last_seen)  # In case it's the constant NEVER_SEEN
+            last_seen = None # In case it's the constant NEVER_SEEN
+
+        info['last_learned_from'] = last_seen
 
-        fleet_icon = node.fleet_state_nickname
-        if fleet_icon is UNKNOWN_FLEET_STATE:
-            fleet_icon = "?"  # TODO  NRN, MN
+        # TODO: what *is* the `timestamp`, anyway? When is it created?
+        info['timestamp'] = self.timestamp.iso8601()
+
+        # TODO: how come we know about the node but not about its fleet state? Does it ever happen?
+        if self.checksum_address in known_nodes.remote_states:
+            info['recorded_fleet_state'] = known_nodes.remote_states[self.checksum_address].to_json()
         else:
-            fleet_icon = fleet_icon.icon
-
-        payload = {"icon_details": node.nickname.payload(),
-                   "rest_url": node.rest_url(),
-                   "nickname": str(node.nickname),
-                   "worker_address": node.worker_address,
-                   "staker_address": node.checksum_address,
-                   "timestamp": node.timestamp.iso8601(),
-                   "last_seen": last_seen,
-                   "fleet_state": node.fleet_state_checksum or 'unknown',
-                   "fleet_state_icon": fleet_icon,
-                   "domain": node.domain,
-                   'version': nucypher.__version__
-                   }
-        return payload
+            info['recorded_fleet_state'] = None
 
-    def abridged_node_details(self, raise_invalid=True) -> dict:
-        """Self-Reporting"""
-        payload = self.node_details(node=self)
-        states = self.known_nodes.abridged_states_dict()
-        known = self.known_nodes_details(raise_invalid=raise_invalid)
-        payload.update({'states': states, 'known_nodes': known})
-        if not self.federated_only:
-            payload.update({
-                "balances": dict(eth=float(self.eth_balance), nu=float(self.token_balance.to_tokens())),
-                "missing_commitments": self.missing_commitments,
-                "last_committed_period": self.last_committed_period})
-        return payload
+        return info
+
+    def status_info(self, raise_invalid=True):
+        # FIXME: is anyone using `raise_invalid=True`? Or is it always `False`?
+
+        for node in self.known_nodes:
+            node.mature()
+
+        info = self._status_info_base()
+
+        info['domain'] = self.domain
+        info['version'] = nucypher.__version__
+
+        latest_states = self.known_nodes.latest_states(5)
+
+        info['fleet_state'] = latest_states[-1].to_json()
+        info['previous_fleet_states'] = [state.to_json() for state in latest_states[:-1]]
+        info['known_nodes'] = [node._status_info_remote(self.known_nodes, raise_invalid=raise_invalid)
+                               for node in self.known_nodes.values()]
```

Comment:
I know this wasn't introduced by this PR, but how about adding a parameter here so that you can get the status info with/without the known_nodes data? That would be particularly useful if we also enable such parameters in the `/status` endpoint. @vepkenez and I were working on some potential crawling logic that checks nodes status information and the known_nodes is just too much data, which in most cases it's not needed.

Reply:
Added an `omit_known_nodes` optional parameter - not sure about the name though.